packages/bundle-size/src/commands/uploadReport.js

```diff
@@ -0,0 +1,139 @@
+const { odata, TableClient, TablesSharedKeyCredential, TableTransaction } = require('@azure/data-tables');
+const chalk = require('chalk');
+
+const collectLocalReport = require('../utils/collectLocalReport');
+const { hrToSeconds } = require('../utils/helpers');
+
+/** @typedef {import('../utils/collectLocalReport').BundleSizeReportEntry} BundleSizeReportEntry */
+
+const AZURE_STORAGE_ACCOUNT = 'fluentbundlesize';
+const AZURE_STORAGE_TABLE_NAME = 'latest';
+const AZURE_ACCOUNT_KEY = process.env.BUNDLESIZE_ACCOUNT_KEY;
+
+/**
+ *  @param {BundleSizeReportEntry} entry
+ *
+ * @return {string}
+ */
+function createRowKey(entry) {
+  // Azure does not support slashes in "rowKey"
+  // https://docs.microsoft.com/archive/blogs/jmstall/azure-storage-naming-rules
+  return `${entry.packageName}${entry.path.replace(/.fixture.js$/, '').replace(/\//g, '')}`;
+}
+
+/**
+ * @param {typeof import('../index') & { branch: string, 'commit-sha': string }} options
+ */
+async function uploadReport(options) {
+  if (!process.env.CI) {
+    console.log(`${chalk.red('[e]')} This is command can be executed only in CI`);
+    process.exit(1);
+  }
+
+  const { branch, 'commit-sha': commitSHA, quiet } = options;
+  const startTime = process.hrtime();
+
+  const localReportStartTime = process.hrtime();
+  const localReport = await collectLocalReport();
+
+  if (!quiet) {
+    console.log(
+      [chalk.blue('[i]'), `Local report prepared in ${hrToSeconds(process.hrtime(localReportStartTime))}`].join(' '),
+    );
+  }
+
+  if (typeof AZURE_ACCOUNT_KEY === 'undefined') {
+    console.log(
+      [
+        chalk.red('[e]'),
+        'process.env.BUNDLESIZE_ACCOUNT_KEY is not defined, please verify Azure Pipelines settings',
+      ].join(' '),
+    );
+    process.exit(1);
+  }
+
+  const credentials = new TablesSharedKeyCredential(AZURE_STORAGE_ACCOUNT, AZURE_ACCOUNT_KEY);
+  const client = new TableClient(
+    `https://${AZURE_STORAGE_ACCOUNT}.table.core.windows.net`,
```

Comment:
can we move this to some standard configurable format outside from javascript ? json for example

Reply:
https://github.com/microsoft/fluentui/pull/18322#discussion_r657806360